src/observability_utils/tracing/decorators.py

```diff
@@ -0,0 +1,192 @@
+"""A set of decorators and associated functions to smooth the experience of
+using the raw opentelemetry decorators and context propagation functions"""
+
+import functools
+from _collections_abc import Callable
+from inspect import signature
+from typing import Any, Concatenate, TypeVar
+
+from opentelemetry.context import attach
+from opentelemetry.propagate import get_global_textmap
+from opentelemetry.trace import SpanKind, Tracer
+from typing_extensions import ParamSpec
+
+T = TypeVar("T")
+P = ParamSpec("P")
+SEPARATOR = "."
```

Comment:
This constant should probably be private

```suggestion
_SEPARATOR = "."
```

Reply:
Done